mayastor/src/bdev/nexus/nexus_bdev.rs

```diff
@@ -633,6 +651,25 @@ impl Nexus {
         Ok(())
     }
 
+    /// set ANA state of the NVMe subsystem
+    pub async fn set_ana_state(
+        &self,
+        ana_state: NvmeAnaState,
+    ) -> Result<(), Error> {
```

Comment:
How about we change this prototype towards Option<NvmeAnaState>? 

1. if we change state from A to B we return A (previous state) 
2. if we change from state A to A (i.e. no change), we return None. 
3. in the case of iSCSI, we return None.

Reply:
This makes it impossible to return `Error`s, which it can now do from `pause`/`resume` as well so I've opted not to do this.